src/main/java/org/minions/devfund/bruno/html/HtmlTag.java

```diff
@@ -0,0 +1,148 @@
+package org.minions.devfund.bruno.html;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Set;
+
+/**
+ * Represents information about a single HTML tag.
+ */
+public class HtmlTag {
+
+    protected final String element;
+    protected final boolean openTag;
+
+    /**
+     * Initializes an instance of {@link HtmlTag}.
+     *
+     * @param element   element name.
+     * @param isOpenTag
+     */
+    public HtmlTag(final String element, final boolean isOpenTag) {
+        this.element = element.toLowerCase();
+        openTag = isOpenTag;
+    }
+
+    /**
+     * Gets the element name (String) specified in this tag.
+     *
+     * @return the element name.
+     */
+    public String getElement() {
+        return element;
+    }
+
+    /**
+     * Checks whether this is the opening tag.
+     *
+     * @return false if is open the tag.
+     */
+    public boolean isOpenTag() {
+        return openTag && !isSelfClosing();
+    }
+
+    /**
+     * Checks whether an HtmlTag other is the matching open/close tag to itself.
+     *
+     * @param other Html tag
+     * @return
+     */
+    public boolean matches(final HtmlTag other) {
+        return other != null
+                && element.equalsIgnoreCase(other.element)
+                && openTag != other.openTag;
+    }
+
+    /**
+     * Checks whether a tag is self-closing.
+     *
+     * @return if is self closing return true.
+     */
+    public boolean isSelfClosing() {
+        return SELF_CLOSING_TAGS.contains(element);
+    }
+
+    public boolean equals(Object obj) {
+        if (obj instanceof HtmlTag) {
+            HtmlTag other = (HtmlTag) obj;
+            return element.equals(other.element)
+                    && openTag == other.openTag;
+        }
+        return false;
+    }
+
+    public String toString() {
+        return "<".concat(openTag ? "" : "/")
+                .concat(element.equals("!--") ? "!-- --" : element).concat(">");
+    }
+
+    /**
+     * The remaining fields and functions are related to HTML file parsing.
+     */
+
+    // a set of tags that don't need to be matched (self-closing)
+    protected static final Set<String> SELF_CLOSING_TAGS = new HashSet<String>(
+            Arrays.asList("!doctype", "!--", "?xml", "xml", "area", "base",
+                    "basefont", "br", "col", "frame", "hr", "img",
+                    "input", "link", "meta", "param"));
+
+
+    protected static final String WHITESPACE = " \f\n\r\t";
+
+    public static Queue<HtmlTag> tokenize(String text) {
+        StringBuffer buf = new StringBuffer(text);
+        Queue<HtmlTag> queue = new LinkedList<HtmlTag>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)